### PR TITLE
renamed extraction setting objects

### PR DESCRIPTION
**Pull request by MaxBenChrist**
To make them more catchy and to better reflect their behavior,  I picked `ExtendedParaMap`, `RecommendedParaMap` and `MinimalParaMap`

### CODE REVIEW

File: docs/text/feature_extraction_settings.rst [hunk 1]
```diff
@@ -12,6 +12,216 @@ After digging deeper into your data, you maybe want to calculate more of a certa
...
+>>> from tsfresh.feature_extraction import ExtendedParaMap
+>>> settings = FeatureExtractionSettings()
+>>> # Set here the options of the settings object as shown in the chapters below
+>>> # ...
+>>> from tsfresh.feature_extraction import extract_features
+>>> extract_features(df, default_calculation_settings_mapping=settings)
+
+
+The
+feature calculators in this package).
```
Comment (jneuff) [inline docs/text/feature_extraction_settings.rst]: Trailing ')'

File: docs/text/feature_extraction_settings.rst [hunk 2]
```diff
@@ -12,6 +12,216 @@ After digging deeper into your data, you maybe want to calculate more of a certa
...
+
+After digging deeper into your data, you maybe want to calculate more of a certain type of feature and less of another
+type. So, you need to use custom settings for the feature extractors. To do that with tsfresh you will have to use a
+custom settings object, like shown now:
+
```
Comment (jneuff) [inline docs/text/feature_extraction_settings.rst]: You repeat this paragraph a couple of times.

**Comment by jneuff** [comment]
The names `Recommended-` and `Minimal-` are good. But in my opinion `Extended-` should rather be `Full-` or `Complete-`. Also, I think it would be easier to read and understand if they were called `-ParameterMap`.

**Comment by MaxBenChrist** [comment]
Complete or full implies that we calculated all possible features, which is wrong. We could calculate infinitely many features by alternating the parameters. I would stick with extended.

Von meinem iPhone gesendet

File: docs/text/feature_extraction_settings.rst (hunk 2 above)
Comment (MaxBenChrist) [inline docs/text/feature_extraction_settings.rst]: Just want to make sure that people get the message ;)

**Comment by MaxBenChrist** [comment]
Before we merge this, we still have to redo the documentation. What do you think about the naming @nils-braun ?

**Comment by nils-braun** [comment]
@MaxBenChrist I am with @jneuff for naming it `ParameterMap` instead of the shorter but more unreadable version.
Concerning the prefix, this is fine for me. I am a bit worried about `Recommended`, because who will want to use a not-recommended parameter map (which implies the naming)...

**Comment by MaxBenChrist** [comment]
> @MaxBenChrist I am with @jneuff for naming it ParameterMap instead of the shorter but more unreadable version.
Concerning the prefix, this is fine for me. I am a bit worried about Recommended, because who will want to use a not-recommended parameter map (which implies the naming)...

I think most users should use the rommended settings. The extended settings are slower and only contain 2 features more at this point.

**Comment by MaxBenChrist** [comment]
In most cases I checked, it slowed down the process significantly.

**Comment by nils-braun** [comment]
Yes, I am with you. It's just about the naming. I am not sure if Recommend Settings is good, because it implies the other settings should never be used (because they are not recommended)

**Comment by MaxBenChrist** [comment]
Maybe _efficient_ or _rational_ are better wordings instead of recommended? :D

**Comment by MaxBenChrist** [comment]
I will merge this pr into #138 so we only have one pr

**Pull request closed without merging**